mesonbuild/compilers/mixins/gnu.py

```diff
@@ -337,3 +337,19 @@ def get_pch_suffix(self) -> str:
 
     def openmp_flags(self) -> typing.List[str]:
         return ['-fopenmp']
+
+    def has_arguments(self, args, env, code, mode):
+        # For some compiler command line arguments, the GNU compilers will
+        # emit a warning on stderr indicating that an option is valid for a
+        # another language, but still complete with exit_success
+        with self._build_wrapper(code, env, args, None, mode, disable_cache=False, want_output=True) as p:
+            result = p.returncode == 0
+            try:
+                stdout, stderr = p.communicate()
+                if self.language == "cpp" and "is valid for C/ObjC" in stderr.decode():
```

Comment:
this should probably be `if self.language in {'cpp', 'objcpp'}`

Reply:
Indeed.  Appreciate all the extra help.  I also found a better way to get the error message so I could drop the try - except clause.  I think this also could be a change brought in on my latest rebase.